cmake/FindSCIPY_OPENBLAS32.cmake

```diff
@@ -0,0 +1,40 @@
+###############################################################################################
+# This file provides macros to process CUDA, CUDAToolkit, cuQuantum and MPI external libraries.
+###############################################################################################
+
+# Include this only once
+include_guard()
+
+macro(get_scipy_openblas_libname SCIPY_OPENBLAS_LIB_NAME)
+    set(LIB_SUFFIX ".so")
+    if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+        set(LIB_SUFFIX ".dylib")
+    elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
+        set(LIB_SUFFIX ".dll")
+    endif()
+
+    set(${SCIPY_OPENBLAS_LIB_NAME} "libscipy_openblas${LIB_SUFFIX}")
+endmacro()
+
+macro(find_path_to_openblas SCIPY_OPENBLASE_LIB_PATH)
+    set(SCIPY_OPENBLAS_LIB_NAME "")
+    get_scipy_openblas_libname(SCIPY_OPENBLAS_LIB_NAME)
+
+    find_file(SCIPY_OPENBLAS_LIB_FILE
+        NAMES   ${SCIPY_OPENBLAS_LIB_NAME}
+        HINTS   ${SCIPY_OPENBLAS}/lib
+                $ENV{SCIPY_OPENBLAS32}/lib
+                ${Python_SITELIB}/scipy_openblas32/lib
+                ${SCIPY_OPENBLAS}
+                $ENV{SCIPY_OPENBLAS32}
+                ENV LD_LIBRARY_PATH
+    )
+
+    if(NOT SCIPY_OPENBLAS_LIB_FILE)
+        message(FATAL_ERROR "\nUnable to find ${SCIPY_OPENBLAS_LIB_NAME}. Please set a SCIPY_OPENBLAS32 env variable to provide the path to scipy_openblas32.")
```

Comment:
Does it work if users specify the path to `scipy/openblas` instead of `scipy_openblas32`?

Reply:
Not really. Current C++ implementation will load `libscipy_openblas.so` for the runpath. However, the sonames of libs in the scipy.libs vary from version to verion, platform to platform.